net-utils/src/lib.rs

```diff
@@ -605,19 +606,35 @@ pub fn bind_two_in_range_with_offset_and_config(
 ///
 /// This will start at a random point in the range provided, and search sequenctially.
 /// If it can not find anything, an Error is returned.
+///
+/// Keep in mind this will not reserve the port for you, only find one that is empty.
 pub fn find_available_port_in_range(ip_addr: IpAddr, range: PortRange) -> io::Result<u16> {
+    find_available_ports_in_range(ip_addr, range, 1).map(|v| *v.first().unwrap())
+}
+
+/// Searches for several ports on a given binding ip_addr in the provided range.
+///
+/// This will start at a random point in the range provided, and search sequencially.
+/// If it can not find anything, an Error is returned.
+pub fn find_available_ports_in_range(
+    ip_addr: IpAddr,
+    range: PortRange,
+    mut num: usize,
+) -> io::Result<Vec<u16>> {
```

Comment:
&lt;bikeshedding&gt;

With const generics, you can write a slightly nicer API for this function.
Like this:

```suggestion
pub fn find_available_ports_in_range<const N: usize>(
    ip_addr: IpAddr,
    range: PortRange,
) -> io::Result<[u16; N]> {
```

The difference is that the caller can selects the number of ports they want using a pattern match:

```rust
    let [port1, port2, port3] =
        find_available_ports_in_range(addr, (1024, 65535))
            .expect("There should be at least 3 available ports");
```

The compiler will infer the value of `N` from the call site.

Reply:
cool idea, yes, I was about to bump major version on this crate anyway, will incorporate this!